lib/Dialect/TTNN/IR/TTNNOps.cpp

```diff
@@ -121,6 +121,42 @@ ::mlir::LogicalResult mlir::tt::ttnn::MatmulOp::verify() {
 }
 // ANCHOR_END: adding_an_op_matmul_ttnn_verify
 
+::mlir::LogicalResult mlir::tt::ttnn::Conv2dOp::verify() {
+  ::mlir::RankedTensorType inputType = getInput().getType();
+  ::mlir::RankedTensorType weightType = getWeight().getType();
+  ::mlir::RankedTensorType biasType = getBias().getType();
+  // ::mlir::RankedTensorType outputType = getOutput().getType();
+  auto inputShape = inputType.getShape();
+  auto weightShape = weightType.getShape();
+  auto biasShape = biasType.getShape();
+  // auto outputShape = outputType.getShape();
+  if (inputShape.size() < 3) {
+    return emitOpError("Input must be at least a 3D tensor");
```

Comment:
Not related to this change, but this makes me think that having a canonicalizer that makes sure there's batch size in ops' tensors would be useful. @nsmithtt thoughts?

Reply:
One of the goals of tensor layout is not to enforce a particular tensor rank.  If you mean specifically for the ttnn dialect though (not ttir) then that could be useful. I think they automatically snap to 4d in a lot of cases.